src/services/ExposureNotificationService/ExposureConfigurationValidator.ts

```diff
@@ -0,0 +1,25 @@
+import {Schema, Validator, ValidatorResult} from 'jsonschema';
+
+import {ExposureConfiguration} from '../../bridge/ExposureNotification';
+
+export class ExposureConfigurationValidationError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'ExposureConfigurationValidationError';
+  }
+}
+
+export class ExposureConfigurationValidator {
+  validateExposureConfiguration(exposureConfiguration: ExposureConfiguration, schema: Schema): ValidatorResult {
+    const validator = new Validator();
+    const validatorResult = validator.validate(exposureConfiguration, schema);
+    if (!validatorResult.valid) {
+      console.log('invalid json');
+      console.log(validatorResult.errors.toString());
+      throw new ExposureConfigurationValidationError(
```

Comment:
Will this cause a fallback to the default?

Reply:
Let's add some test cases for some "bad" configs

Reply:
Yes, it will cause a fallback to either using a previously saved or the default config. This thrown error is caught in `performExposureStatusUpdate`, in *ExposureNotificationService.ts*